### PR TITLE
Improve user search display name

### PR DESCRIPTION
**Pull request by KN4CK3R**
I tripped over this strange method and I don't think we need that workaround to fix the value.

old:
grafik

new:
grafik

### CODE REVIEW

File: web_src/js/features/repo-settings.js
Comment (delvh) [inline web_src/js/features/repo-settings.js]:
```suggestion
          items.push({
```
Seems like an oversight.
(Regarding why two lines: Sometimes I hate GitHub. They introduced yet again a reviewing bug.)

**Review by delvh — APPROVED** [review]

File: web_src/js/features/repo-settings.js
Comment (KN4CK3R) [reply web_src/js/features/repo-settings.js]: What are you suggesting here?

**Review by silverwind — APPROVED** [review]

**Review by wxiaoguang — APPROVED** [review]
XSS (approved by mistake, add a new change request)

Update: resolved.

<details>

image

image

</details>

**Review by wxiaoguang — DISMISSED** [review]
XSS (feel free to dismiss this change request if the concern has been addressed)

**Comment by KN4CK3R** [comment]
That's strange because Fomantic escapes the text...
https://github.com/fomantic/Fomantic-UI/blob/13b76ee714bba8efdeef8a6e402cda6774abce40/src/definitions/modules/search.js#L1493-L1498

**Comment by wxiaoguang** [comment]
> That's strange because Fomantic escapes the text... https://github.com/fomantic/Fomantic-UI/blob/13b76ee714bba8efdeef8a6e402cda6774abce40/src/definitions/modules/search.js#L1493-L1498

No. See `preserveHTML`

**Comment by wxiaoguang** [comment]
Maybe there are some nits. eg: the misalignment (when the item has "description") (since it has got lgtm/done, so this comment is just FYI, no blocker)

**Comment by silverwind** [comment]
Right, the avatar is misalign. Retracting review for now. Likely needs a CSS fix.

**Comment by KN4CK3R** [comment]
That's just the default style of Fomantic. Feel free to change whatever CSS is needed.

File: web_src/js/features/repo-settings.js
Comment (delvh) [reply web_src/js/features/repo-settings.js]: Removing the variable

**Comment by silverwind** [comment]
I will take a look later. Likely it's incorrect/unsuppored usage of fomantic that's causing the misalignment.

File: web_src/js/features/repo-settings.js
Comment (KN4CK3R) [reply web_src/js/features/repo-settings.js]: Already removed, thanks.

**Comment by silverwind** [comment]
CSS fixes done:

image

**Review by silverwind — APPROVED** [review]

File: web_src/css/repo.css
Comment (wxiaoguang) [inline web_src/css/repo.css]: Is the comment above `/* this trick is used to align with the sibling avatar image */` still correct?

Comment (silverwind) [inline web_src/css/repo.css]: Side note: Maybe it's better to make this declaration "global" later. Most of the UI should be able to deal correctly with content wrapping.

File: web_src/js/features/comp/SearchUserBox.js
Comment (wxiaoguang) [inline web_src/js/features/comp/SearchUserBox.js]:
```suggestion
          if (item.full_name) {
            resultItem.description = htmlEscape(item.full_name);
          }
```

`if (item.full_name)` already implies `item.full_name.length > 0`

Or maybe even simpler: remove the `if` check (not sure whether Fomantic works with `description = null`)

File: web_src/css/repo.css
Comment (silverwind) [reply web_src/css/repo.css]: No, it's unrelated, without this a unbroken description line would overflow the box:

image

File: web_src/css/base.css
Comment (wxiaoguang) [inline web_src/css/base.css:206]: Are there other `.ui.search` components? Does this changed layout work with all of them?

File: web_src/css/repo.css
Comment (silverwind) [reply web_src/css/repo.css]: This is because fomantic's `.ui.search > .results { width: 18em; }` so I guess I will move this to the same selector.

Comment (wxiaoguang) [reply web_src/css/repo.css]: I mean, if the comment is not correct anymore, does it need to be updated?

Comment (silverwind) [reply web_src/css/repo.css]: Selector moved.

Comment (silverwind) [reply web_src/css/repo.css]: Comment is not accurate anymore, but the rule is still needed to avoid strange rendering, I will just remove the comment.

File: web_src/css/base.css
Comment (silverwind) [reply web_src/css/base.css:206]: Yes there are more, but likely it won't break them. Should test them though.

Comment (silverwind) [reply web_src/css/base.css:206]: Actually maybe this is the only case where there are `.results`.

File: web_src/css/repo.css
Comment (wxiaoguang) [inline web_src/css/repo.css]: According to my test, I think it should be:

```
.ui.search > .results .result .image + .content {
    margin: 0; /* reset Fomantic's margin because we have `align-items: center;` on the `.result` parent
}
```

Comment (silverwind) [reply web_src/css/repo.css]: Not quite, correct comment in https://github.com/go-gitea/gitea/pull/29002/commits/6da06851c76a0358c4063d58f2e0cedb283000b1.

File: web_src/js/features/comp/SearchUserBox.js
Comment (silverwind) [reply web_src/js/features/comp/SearchUserBox.js]: Probably better to not try with `null`.

File: web_src/js/features/repo-settings.js
Comment (silverwind) [inline web_src/js/features/repo-settings.js]:
```suggestion
            description: `${item.permission} access` // TODO: translate this string
```

**Review by wxiaoguang — APPROVED** [review]

File: web_src/css/repo.css
Comment (silverwind) [inline web_src/css/repo.css]:
```suggestion
  margin: 0; /* remove margin reserved for avatar because we move it to left via `order: 0` */
```

**Comment by KN4CK3R** [comment]
Thank you for adjusting the styles. 👍

**Pull request merged**
